src/openldap/mod.rs

```diff
@@ -0,0 +1,517 @@
+    /// Configuring the admin for the access log can be done via [`OpenLDAP::with_accesslog_admin`]
+    /// Default: `None`
+    pub fn with_accesslog_settings(
+        mut self,
+        AccesslogSettings {
+            log_operations,
+            log_success,
+            log_purge,
+            log_old,
+            log_old_attribute,
+        }: AccesslogSettings,
+    ) -> Self {
+        self.env_vars
+            .insert("LDAP_ENABLE_ACCESSLOG".to_owned(), "yes".to_owned());
+        self.env_vars.insert(
+            "LDAP_ACCESSLOG_LOGOPS".to_owned(),
+            log_operations.to_string(),
+        );
+        if log_success {
+            self.env_vars
+                .insert("LDAP_ACCESSLOG_LOGSUCCESS".to_owned(), "TRUE".to_owned());
+        } else {
+            self.env_vars
+                .insert("LDAP_ACCESSLOG_LOGSUCCESS".to_owned(), "FALSE".to_owned());
+        }
+        self.env_vars.insert(
+            "LDAP_ACCESSLOG_LOGPURGE".to_owned(),
+            format!("{} {}", log_purge.0, log_purge.1),
+        );
+        self.env_vars
+            .insert("LDAP_ACCESSLOG_LOGOLD".to_owned(), log_old.to_owned());
+        self.env_vars.insert(
+            "LDAP_ACCESSLOG_LOGOLDATTR".to_owned(),
+            log_old_attribute.to_owned(),
+        );
+        self
+    }
+
+    /// Activates the access log and sets the admin user up (there can only be one)
+    /// Configuring how [`OpenLDAP`] logs can be done via [`OpenLDAP::with_accesslog_settings`]
+    /// Default: `None`
+    pub fn with_accesslog_admin(mut self, username: &str, password: &str) -> Self {
+        self.env_vars
+            .insert("LDAP_ENABLE_ACCESSLOG".to_owned(), "yes".to_owned());
+        self.env_vars.insert(
+            "LDAP_ACCESSLOG_ADMIN_USERNAME".to_owned(),
+            username.to_owned(),
+        );
+        self.env_vars.insert(
+            "LDAP_ACCESSLOG_ADMIN_PASSWORD".to_owned(),
+            password.to_owned(),
+        );
+        self
+    }
+
+    /// Adds a user (can be called multiple times)
+    /// Default: `[]`
+    ///
+    /// Alternatively, users can be added via [`OpenLDAP::with_users`].
+    pub fn with_user(mut self, username: &str, password: &str) -> Self {
+        self.users.push(User {
+            username: username.to_owned(),
+            password: password.to_owned(),
+        });
+        self
+    }
+
+    /// Add users (can be called multiple times)
+    /// Default: `[]`
+    ///
+    /// Alternatively, users can be added via [`OpenLDAP::with_user`].
+    pub fn with_users<Username: Into<String> + Clone, Password: Into<String> + Clone>(
+        mut self,
+        user_password: &[(Username, Password)],
+    ) -> Self {
+        for (username, password) in user_password {
+            self.users.push(User {
+                username: username.clone().into(),
+                password: password.clone().into(),
+            })
+        }
+        self
+    }
```

Comment:
Sorry, just realized that it's a slice of tuples. Thus, we can't avoid allocation
In that case, `ToString` may be better than `Into<String> + Clone`

```suggestion
    pub fn with_users<Username: ToString, Password: ToString>(
        mut self,
        user_password: &[(Username, Password)],
    ) -> Self {
        for (username, password) in user_password {
            self.users.push(User {
                username: username.to_string(),
                password: password.to_string(),
            })
        }
        self
    }
```